quadtree/quadkey.go

```diff
@@ -8,18 +8,21 @@ import (
 	"github.com/peterstace/simplefeatures/geom"
 )
 
-// Misc functions for generating/calculating quadkeys
-
+// QuadKey is a key representing a Slippy tile.
 type QuadKey uint64
 
+const MaxZoom = 29
+
+// Zoom level is the bottom 5 bits
+const zoomMask = 0b11111
+
 // Parent get parents quadkey for passed quadkey
 func (q QuadKey) Parent() (QuadKey, error) {
 	zoomLevel := q.Zoom()
-	parentZoomLevel := zoomLevel - 1
-
-	if parentZoomLevel <= 0 {
+	if zoomLevel <= 0 {
 		return 0, errors.New("no parent")
 	}
+	parentZoomLevel := zoomLevel - 1
```

Comment:
ummm that will allow ParentZoomLevel to be -1.

Reply:
less than or equal to

Reply:
It might be clearer to keep the original and change it to
```go
if parentZoomLevel <= 0 {
```

Reply:
Ah, I knew there was a reason I did it this way. `byte` is unsigned, so
```go
if parentZoomLevel < 0 {
```
is trivially false.